_drafts/2020-09-17-fixing-an-n1-performance-problem-in-xafxpo-with-totally-undocumented-apis.md

```diff
@@ -0,0 +1,48 @@
+---
+ layout: post 
+ title: Fixing an N+1 performance problem in XAF/XPO with totally undocumented APIs
+ comments: true
+ github: fixing-an-n-plus-1-perf-problem-in-xaf-xpo
+ tags: ["XAF", "XPO", "Performance", "BestPractices", "ORM"]
+---
+
+We all know (or not) that using an abstraction over any kind of datasource can lead to performance problems. That's nothing new and is quite common with ORM's.
+
+Let's look at the business problem we need to solve first before discussing why N+1 happens and what we can do to avoid it.
+It's a quite common problem with relational databases, an especially when dealing with any ORM. So we need to be careful to avoid it and there are several options we have when dealing with that kind of problem.
```

Comment:
```suggestion
It's quite a common problem with relational databases, and especially when dealing with any ORM. So we need to be careful to avoid it. Luckily there are several options when dealing with that kind of problem.
```